jax/_src/lax/lax.py

```diff
@@ -1016,15 +1016,17 @@ def _get_monoid_reducer(monoid_op: Callable,
   aval = core.get_aval(x)
   dtype = _dtype(x)
   if (type(aval) is ConcreteArray) and aval.shape == ():
+    # allow bitwise reductions for boolean and integer types
+    _is_intlike = dtype == np.bool_ or dtypes.issubdtype(dtype, np.integer)
     if monoid_op is add:
       return np.equal(aval.val, 0) and partial(_reduce_sum)
     elif monoid_op is mul:
       return np.equal(aval.val, 1) and _reduce_prod
-    elif monoid_op is bitwise_or and dtype == np.bool_:
+    elif monoid_op is bitwise_or and _is_intlike:
       return np.equal(aval.val, _get_max_identity(dtype)) and _reduce_or
-    elif monoid_op is bitwise_and and dtype == np.bool_:
+    elif monoid_op is bitwise_and and _is_intlike:
       return np.equal(aval.val, _get_min_identity(dtype)) and _reduce_and
-    elif monoid_op is bitwise_xor and dtype == np.bool_:
+    elif monoid_op is bitwise_xor and _is_intlike:
       return np.equal(aval.val, _get_max_identity(dtype)) and _reduce_xor
```

Comment:
I don't think these are the correct identities for bitwise operations in the case of signed integers. For `bitwise_or` and `bitwise_xor` the identity should be the value where all bits are zero, and for `bitwise_and` the identity should be the value where all bits are `1`. For `bool` and `uint` this corresponds to `get_min_identity` and `get_max_identity`, but for signed ints using a two's complement representation I think you need `0` and `-1`

Reply:
Great catch! I think the potential fix could be either of:
1. Adding more statements to that branch specializing the value compared in `np.equal` depending on whether the integer is signed or unsigned (or it is a boolean), or
2. Explicitly check equality against `dtype(0)` and `dtype(-1)`, which should evaluate to the minimum/maximum identities of uints and bools, and the bitwise identities for signed integers, in which case we would get all three cases with a single line.

What do you think? The second might be elegant, but at the same time a little less readable.

Reply:
Yeah, seems like using `dtype(0)` for `bitwise_or`/`bitwise_xor` and `dtype(-1)` for `bitwise_and` would be the cleanest approach.

Reply:
I am getting errors locally of the form `numpy.dtype object is not callable`. Do you have an idea on how I could work around those? Seems like a simple "cast" of the value like I suggested does not work.

Reply:
Use `np.array(-1, dtype=dtype)`